code/_onclick/ai_onclick.dm

```diff
@@ -204,9 +204,15 @@
 	open_close(user)
 
 /obj/machinery/door/airlock/AICtrlClick(mob/living/silicon/user) // Bolts doors
-
 	if(!ai_control_check(user))
 		return
+
+	if(istype(user, /mob/living/silicon/robot))
+		var/mob/living/silicon/robot/S = user
+		if(S.emagged || do_after(user, 3 SECONDS, target = src, allow_moving = TRUE))
```

Comment:
Ok this code isnt great. Lots of copy paste. Can you make a proc on silicon called something like `can_instant_shock` then override it on AI and borgs.
Then do if(can_instant_shock() || do_after())